components/ProgressBar.js

```diff
@@ -1,14 +1,35 @@
 import styled from "styled-components";
+import {MdOutlineSettingsBackupRestore} from "react-icons/md";
+import {useRef} from "react";
+
+export default function Progress({
+  budget,
+  percentage,
+  progress,
+  addvalue,
+  minusvalue,
+}) {
+  const ref = useRef(null);
+  const ref2 = useRef(null);
+  const ref3 = useRef(null);
+
+  function restoreClick() {
+    ref.current.value = "0€";
+    ref2.current.value = "0%";
+    ref3.current.value = "0";
+  }
 
-export default function Progress({budget, percentage, progress, addvalue}) {
   const managepercentage = percentage;
-  const addFive = budget + addvalue + "€";
+  const addFive = budget + addvalue + minusvalue + "€";
   const Budget = budget + "€";
   return (
     <Progressmain className="progressmain">
-      <Procentage1 value={managepercentage + "%"} />
-      <Procentage2 value={Budget && addFive} readonly />
-      <Progressbar value={progress} max="100"></Progressbar>
+      <Procentage1 ref={ref2} value={managepercentage + "%"} />
+      <Procentage2 ref={ref} value={Budget && addFive} />
+      <Restorebutton onClick={restoreClick}>
```

Comment:
Why are you using `refs` for your values? Shouldn't this be state variables?
Furthermore: The percentage value can be calculated at runtime, so no need to have a separate variable for it.
Maybe we should have a look at your code together. I don't understand what these refs are used for.

Reply:
Ok, then we will have a look at my Code Tomorrow.